draft-ietf-jsonpath-base.md

```diff
@@ -1417,10 +1417,9 @@ a non-empty nodelist. The table also lists the (immediate) subtypes of each type
 | :--                       | :----------------                        | :------                                        |
 | `OptionalNodeOrValueType` | `Node(n)`, `Value(v)`, `Nothing`         | `OptionalNodeType`, `OptionalValueType`        |
 | `OptionalNodeType`        | `Node(n)`, `Nothing`                     |                                                |
-| `OptionalValueType`       | `Value(v)`, `Nothing`                    | `OptionalNodeType`, `ValueType`, `OptionalBooleanType` |
+| `OptionalValueType`       | `Value(v)`, `Nothing`                    | `OptionalNodeType`, `ValueType`                |
 | `ValueType`               | `Value(v)`                               | `BooleanType`                                  |
-| `OptionalBooleanType`     | `Value(true)`, `Value(false)`, `Nothing` | `BooleanType`                                  |
-| `BooleanType`             | `Value(true)`, `Value(false)`            |                                                |
+| `BooleanType`             | `Logic`                                  |                                                |
```

Comment:
1. `BooleanType` should be renamed to `PropositionType` to further distance the term from the JSON `true` and `false` values which people (and RFC 8259, in the introduction) sometimes refer to as booleans. We should also rename `boolean-expr` to `proposition` in the ABNF to emphasise the correspondence.

2. There need to be two abstract instances rather than one. If we adopt the renaming above, the abstract instances could be  `PropositionTrue` and `PropositionFalse`.

Reply:
I'm happy with this change. I can do that shortly.

Is there evidence of the term "proposition" used in this way elsewhere? Will this term be understood?

Reply:
The term is often used in the context of proposition logic (or [propositional calculus](https://en.wikipedia.org/wiki/Propositional_calculus)), but it's also quite common in CompSci. The term won't necessarily be familiar to all programmers, but at least from their perspective, "proposition" would then be just an unfamiliar but distinct term in the spec.

Reply:
We could call it LogicalFalse.  But we don't need it!